Lib/fontTools/cffLib/__init__.py

```diff
@@ -271,8 +271,17 @@ def convertCFFToCFF2(self, otFont):
 						# print "Removing privateDict attr", key
 		else:
 			# clean up the PrivateDicts in the fdArray
+			fdArray = topDict.FDArray
 			privateOpOrder = buildOrder(privateDictOperators2)
 			for fontDict in fdArray:
+				fontDict.order = fontDict.orderCFF2
+				fontDict._isCFF2 = True
+				for key in fontDict.rawDict.keys():
+					if key not in fontDict.order:
+						del fontDict.rawDict[key]
+						if hasattr(fontDict, key):
+							exec("del fontDict.%s" % (key))
```

Comment:
@readroberts `exec()` is bad. Use `delattr()` instead https://docs.python.org/3/library/functions.html#delattr